ansible/pmm2/main.yml

```diff
@@ -0,0 +1,427 @@
+        line: 'TELEMETRY_OS=OVF'
+
+    - name: cloud-init                 | Set TELEMETRY_OS
+      when: ansible_virtualization_type == "xen"
+      lineinfile:
+        path: /etc/sysconfig/pmm-managed
+        regexp: '^TELEMETRY_OS='
+        line: 'TELEMETRY_OS=AMI'
+
+    - name: cloud-init                 | Set TELEMETRY_OS
+      when: ansible_virtualization_type == "kvm"
+      lineinfile:
+        path: /etc/sysconfig/pmm-managed
+        regexp: '^TELEMETRY_OS='
+        line: 'TELEMETRY_OS=GCP'
+
+    - name: PMM                        | Fix nginx config
+      replace:
+        dest: /etc/nginx/nginx.conf
+        regexp: '^(\s*)listen'
+        replace: '\1#listen'
+
+    - name: NGINX SSL Certificate      | Add script
+      copy:
+        content: |
+          #!/bin/sh
+
+          set -o errexit
+
+          mkdir -p /srv/nginx
+
+          if [ ! -e /srv/nginx/dhparam.pem ]; then
+              openssl dhparam -out /srv/nginx/dhparam.pem 2048
+          fi
+
+          if [ ! -e /srv/nginx/ca-certs.pem ]; then
+              # Fetch letsencrypt chain
+              curl https://letsencrypt.org/certs/isrgrootx1.pem                          > /srv/nginx/ca-certs.pem
+              curl https://letsencrypt.org/certs/lets-encrypt-x1-cross-signed.pem       >> /srv/nginx/ca-certs.pem
+              curl https://letsencrypt.org/certs/letsencryptauthorityx1.pem             >> /srv/nginx/ca-certs.pem
+              curl https://www.identrust.com/certificates/trustid/root-download-x3.html >> /srv/nginx/ca-certs.pem
+          fi
+
+          if [ ! -e /srv/nginx/certificate.conf ]; then
+              echo "
+          [ req ]
+          distinguished_name = req_distinguished_name
+          prompt             = no
+
+          [ req_distinguished_name ]
+          O                  = Main Org.
+          " > /srv/nginx/certificate.conf
+          fi
+
+          if [ ! -e /srv/nginx/certificate.key -o ! -e /srv/nginx/certificate.crt ]; then
+              openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+                  -keyout /srv/nginx/certificate.key \
+                  -out    /srv/nginx/certificate.crt \
+                  -config /srv/nginx/certificate.conf
+          fi
+        dest: /var/lib/cloud/scripts/per-boot/generate-ssl-certificate
+        mode: 0755
+
+    - name: NGINX SSL Certificate      | Check certificate file
+      stat: path=/srv/nginx/certificate.crt
+      register: certificate_file
+
+    - name: NGINX SSL Certificate      | Generate certificate
+      when: not certificate_file.stat.exists
+      command: /var/lib/cloud/scripts/per-boot/generate-ssl-certificate
+
+    - name: Grafana                    | Check data dir
+      stat: path=/srv/grafana/grafana.db
+      register: grafana_db
+
+    - name: PMM                        | Fix path to grafana.db
+      command: sh -c 'sed -i "s|/var/lib/grafana/grafana.db|/srv/grafana/grafana.db|" /etc/supervisord.d/pmm.ini'
```

Comment:
why not fix it in pmm-server repo directly?

Reply:
I will create a separate task for these changes and as soon as it is fixed will change this code.